### PR TITLE
extern C each function individually

### PR DESCRIPTION
**Pull request by KarthikRIyer**

### CODE REVIEW

File: include/copentimelineio/anyVector.h [hunk 1]
```diff
@@ -4,51 +4,78 @@
...
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+struct AnyVectorIterator;
+typedef struct AnyVectorIterator AnyVectorIterator;
+struct AnyVector;
+typedef struct AnyVector AnyVector;
+struct AnyVectorMutationStamp;
+typedef struct AnyVectorMutationStamp AnyVectorMutationStamp;
```
Comment (meshula) [inline include/copentimelineio/anyVector.h:16]:
```suggestion
typedef struct AnyVectorIterator AnyVectorIterator;
typedef struct AnyVector AnyVector;
typedef struct AnyVectorMutationStamp AnyVectorMutationStamp;
```

File: include/copentimelineio/composableRetainerVector.h [hunk 2]
```diff
@@ -4,63 +4,86 @@
...
 #ifdef __cplusplus
-extern "C"
-{
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
-    struct ComposableRetainerVectorIterator;
-    typedef struct ComposableRetainerVectorIterator
+struct ComposableRetainerVectorIterator;
```
Comment (meshula) [inline include/copentimelineio/composableRetainerVector.h:11]:
```suggestion
struct ComposableRetainerVectorIterator;
```

File: include/copentimelineio/composableRetainerVector.h [hunk 3]
```diff
@@ -4,63 +4,86 @@
...
-        ComposableRetainerVectorIterator* rhs);
-    bool ComposableRetainerVectorIterator_not_equal(
-        ComposableRetainerVectorIterator* lhs,
-        ComposableRetainerVectorIterator* rhs);
-    void ComposableRetainerVectorIterator_destroy(
-        ComposableRetainerVectorIterator* self);
-#ifdef __cplusplus
-}
-#endif
+struct ComposableRetainerVector;
```
Comment (meshula) [inline include/copentimelineio/composableRetainerVector.h:14]:
```suggestion
```

File: include/copentimelineio/effect.h [hunk 4]
```diff
@@ -4,37 +4,50 @@
...
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+
+struct RetainerEffect;
+typedef struct RetainerEffect RetainerEffect;
+struct Effect;
+typedef struct Effect Effect;
+
```
Comment (meshula) [inline include/copentimelineio/effect.h:16]:
```suggestion
typedef struct RetainerEffect RetainerEffect;
typedef struct Effect Effect;

```

File: include/copentimelineio/effectRetainerVector.h [hunk 5]
```diff
@@ -4,58 +4,83 @@
...
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+
+struct EffectRetainerVectorIterator;
+typedef struct EffectRetainerVectorIterator EffectRetainerVectorIterator;
+struct EffectRetainerVector;
+typedef struct EffectRetainerVector EffectRetainerVector;
```
Comment (meshula) [inline include/copentimelineio/effectRetainerVector.h:15]:
```suggestion
typedef struct EffectRetainerVectorIterator EffectRetainerVectorIterator;
typedef struct EffectRetainerVector EffectRetainerVector;
```

File: include/copentimelineio/effectVector.h [hunk 6]
```diff
@@ -4,48 +4,72 @@
...
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+struct EffectVectorIterator;
+typedef struct EffectVectorIterator EffectVectorIterator;
+struct EffectVector;
+typedef struct EffectVector EffectVector;
+
```
Comment (meshula) [inline include/copentimelineio/effectVector.h:15]:
```suggestion
typedef struct EffectVectorIterator EffectVectorIterator;
typedef struct EffectVector EffectVector;

```

File: include/copentimelineio/freezeFrame.h [hunk 7]
```diff
@@ -5,34 +5,46 @@
...
-    const char* FreezeFrame_schema_name(FreezeFrame* self);
-    int         FreezeFrame_schema_version(FreezeFrame* self);
-#ifdef __cplusplus
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+
+struct FreezeFrame;
```
Comment (meshula) [inline include/copentimelineio/freezeFrame.h:13]:
```suggestion
```

File: include/copentimelineio/linearTimeWarp.h [hunk 8]
```diff
@@ -5,39 +5,51 @@
...
-    const char* LinearTimeWarp_schema_name(LinearTimeWarp* self);
-    int         LinearTimeWarp_schema_version(LinearTimeWarp* self);
-#ifdef __cplusplus
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+
+struct LinearTimeWarp;
```
Comment (meshula) [inline include/copentimelineio/linearTimeWarp.h:13]:
```suggestion
```

File: include/copentimelineio/markerRetainerVector.h [hunk 9]
```diff
@@ -4,58 +4,82 @@
...
-#ifdef __cplusplus
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+struct MarkerRetainerVectorIterator;
+typedef struct MarkerRetainerVectorIterator MarkerRetainerVectorIterator;
+struct MarkerRetainerVector;
+typedef struct MarkerRetainerVector MarkerRetainerVector;
```
Comment (meshula) [inline include/copentimelineio/markerRetainerVector.h:14]:
```suggestion
typedef struct MarkerRetainerVectorIterator MarkerRetainerVectorIterator;
typedef struct MarkerRetainerVector MarkerRetainerVector;
```

File: include/copentimelineio/markerVector.h [hunk 10]
```diff
@@ -4,48 +4,72 @@
...
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+struct MarkerVectorIterator;
+typedef struct MarkerVectorIterator MarkerVectorIterator;
+struct MarkerVector;
+typedef struct MarkerVector MarkerVector;
+
```
Comment (meshula) [inline include/copentimelineio/markerVector.h:15]:
```suggestion
typedef struct MarkerVectorIterator MarkerVectorIterator;
typedef struct MarkerVector MarkerVector;

```

File: include/copentimelineio/retainerPairComposable.h [hunk 11]
```diff
@@ -3,20 +3,21 @@
...
-    typedef struct RetainerPairComposable RetainerPairComposable;
 
-    RetainerPairComposable* RetainerPairComposable_create(
-        RetainerComposable* first, RetainerComposable* second);
-    RetainerComposable*
-    RetainerPairComposable_first(RetainerPairComposable* self);
-    RetainerComposable*
-         RetainerPairComposable_second(RetainerPairComposable* self);
-    void RetainerPairComposable_destroy(RetainerPairComposable* self);
+struct RetainerPairComposable;
```
Comment (meshula) [inline include/copentimelineio/retainerPairComposable.h:11]:
```suggestion
```

File: include/copentimelineio/serializableObjectRetainerVector.h [hunk 12]
```diff
@@ -3,82 +3,106 @@
...
 #ifdef __cplusplus
-extern "C"
-{
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
-    struct SerializableObjectRetainerVectorIterator;
-    typedef struct SerializableObjectRetainerVectorIterator
+struct SerializableObjectRetainerVectorIterator;
```
Comment (meshula) [inline include/copentimelineio/serializableObjectRetainerVector.h:10]:
```suggestion
```

File: include/copentimelineio/serializableObjectRetainerVector.h [hunk 13]
```diff
@@ -3,82 +3,106 @@
...
-        SerializableObjectRetainerVectorIterator* rhs);
-    bool SerializableObjectRetainerVectorIterator_not_equal(
-        SerializableObjectRetainerVectorIterator* lhs,
-        SerializableObjectRetainerVectorIterator* rhs);
-    void SerializableObjectRetainerVectorIterator_destroy(
-        SerializableObjectRetainerVectorIterator* self);
-#ifdef __cplusplus
-}
-#endif
+struct SerializableObjectRetainerVector;
```
Comment (meshula) [inline include/copentimelineio/serializableObjectRetainerVector.h:13]:
```suggestion
```

File: include/copentimelineio/trackVector.h [hunk 14]
```diff
@@ -4,48 +4,73 @@
...
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+
+struct TrackVectorIterator;
+typedef struct TrackVectorIterator TrackVectorIterator;
+struct TrackVector;
+typedef struct TrackVector TrackVector;
```
Comment (meshula) [inline include/copentimelineio/trackVector.h:15]:
```suggestion
typedef struct TrackVectorIterator TrackVectorIterator;
typedef struct TrackVector TrackVector;
```

File: include/copentimelineio/typeInfo.h [hunk 15]
```diff
@@ -3,12 +3,5 @@
...
-#ifdef __cplusplus
-extern "C"
-{
-#endif
-    struct TypeInfo;
-    typedef struct TypeInfo TypeInfo;
-#ifdef __cplusplus
-}
-#endif
+struct TypeInfo;
```
Comment (meshula) [inline include/copentimelineio/typeInfo.h:6]:
```suggestion
struct TypeInfo;
```

File: include/copentimelineio/unknownSchema.h [hunk 16]
```diff
@@ -4,29 +4,38 @@
...
-    const char* UnknownSchema_schema_name(UnknownSchema* self);
-    int         UnknownSchema_schema_version(UnknownSchema* self);
-#ifdef __cplusplus
-}
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
+
+struct UnknownSchema;
```
Comment (meshula) [inline include/copentimelineio/unknownSchema.h:12]:
```suggestion
```

**Review by meshula — CHANGES_REQUESTED** [review]
One big change all the way through; eliding the redundant struct forward declaration

Thanks!~

**Comment by KarthikRIyer** [comment]
Done! Thanks for the review.

File: include/copentimelineio/composableRetainerVector.h [hunk 17]
```diff
@@ -4,63 +4,85 @@
...
 #ifdef __cplusplus
-extern "C"
-{
+# define OTIO_API extern "C"
+#else
+# define OTIO_API
 #endif
-    struct ComposableRetainerVectorIterator;
-    typedef struct ComposableRetainerVectorIterator
+struct ComposableRetainerVectorIterator;
```
Comment (meshula) [inline include/copentimelineio/composableRetainerVector.h:11]:
```suggestion
```

File: include/copentimelineio/typeInfo.h (hunk 15 above)
Comment (meshula) [inline include/copentimelineio/typeInfo.h:6]:
```suggestion
```

**Review by meshula — CHANGES_REQUESTED** [review]
a couple fell through the cracks

**Review by meshula — APPROVED** [review]

**Pull request merged**